packages/syntax-floating-components/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  MutableRefObject,
+  ReactElement,
+  ReactNode,
+  useLayoutEffect,
+  useRef,
+  useMemo,
+} from "react";
+import classnames from "classnames";
+import {
+  Placement,
+  arrow,
+  flip,
+  useFloating,
+  shift,
+  offset,
+} from "@floating-ui/react";
+
+import styles from "./Tooltip.module.css";
+
+const defaultPlacement = "top";
+
+const Tooltip = ({
+  children,
+  offset: offsetValue = 7, // include arrow size which equals sqrt(5^2 + 5^2) = 7.07 ~ 7,
+  open,
+  placement: placementProp,
+  referenceEl,
+  zIndex: zIndexProp = 1100,
+}: {
+  /**
+   * The children to show the tooltip on hover
+   */
+  children: ReactNode;
+  offset?: number;
+  open: boolean;
+  placement?: Placement;
+  /**
+   * The reference of the children element
+   *
+   * "Make sure the external elements are stored in state, not in a ref" (TODO figure out what this means: https://floating-ui.com/docs/react#refs)
```

Comment:
@JFChoe Should we remove this comment? It actually shows up in storybook: 
![image](https://user-images.githubusercontent.com/127199/234412750-9f4a83c1-4488-4993-8832-2a54cb68f5ef.png)

Reply:
Ah yea, I removed any TODO comments